src/feature/AlignHashRocketsFeature.ts

```diff
@@ -0,0 +1,125 @@
+import { IFeature } from '../feature';
+import * as vscode from "vscode";
+import { ILogger } from "../logging";
+import { IAggregateConfiguration } from "../configuration";
+
+class HashRocketLocation {
+  public rocketIndex: number;
+  public preceedingTextIndex: number;
+}
+
+class AlignHashRocketsFormatter implements vscode.OnTypeFormattingEditProvider{
+  private logger: ILogger;
+
+  constructor(logger: ILogger) {
+    this.logger = logger;
+  }
+
+  private extractHashRocketLocation(document: vscode.TextDocument, line: number): HashRocketLocation {
+    if (line < 0 || line > document.lineCount) { return null; }
+
+    var textLine = document.lineAt(line);
+    // Find the first hash-rocket `=>`
+    var hashRocket = textLine.text.indexOf(' =>');
+    if (hashRocket === -1) { return null; }
+
+    // Now find where the preceeding text ends
+    var textIndex = textLine.text.substr(0, hashRocket).trimRight().length;
+    //        key    => value 
+    //           |  | <--- this is hashRocket
+    //           | <--- this is textIndex
+    var result: HashRocketLocation = {
+      rocketIndex: hashRocket + 1,
+      preceedingTextIndex: textIndex
+    };
+
+    return result;
+  }
+
+  public provideOnTypeFormattingEdits(
+    document: vscode.TextDocument,
+    position: vscode.Position,
+    ch: string,
+    options: vscode.FormattingOptions,
+    token: vscode.CancellationToken):
+    Thenable<vscode.TextEdit[]>
+  {
+    //TODO: Check options for tabs vs spaces
```

Comment:
It's part of the document object IIRC, so shouldn't have to go out to options to get that

Reply:
It was more about _inserting_ tabs etc. as right now it only inserts spaces.